projects/form-builder/src/lib/pipes/show-field/show-field.pipe.ts

```diff
@@ -1,20 +1,63 @@
 import {Pipe, PipeTransform} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {map, startWith} from 'rxjs/operators';
 import {CompiledField} from '../../interfaces/compiled-field.interface';
 import {Parser} from '../../utils/parser';
+import {safeEval} from '../../utils/safe-eval';
 
 @Pipe({
   name: 'showField'
 })
 export class ShowFieldPipe implements PipeTransform {
-  transform(fields: CompiledField[], parser: Parser): CompiledField[] {
+  transform(fields: CompiledField[], parser: Parser): Observable<CompiledField[]> {
+
+    return fields.some(field => field.condition)
```

Comment:
We should first filter the fields based on the `filtered` method and then check to see if we need to implement change observables. In the current setup it's possible that a field that's normally filtered out because of a state condition causes us to implement the observable.